addon/components/stacked-vertical-bar-chart.js

```diff
@@ -128,75 +142,99 @@ const StackedVerticalBarChartComponent = ChartComponent.extend(LegendMixin,
+    var posTop, negBottom, stackedValues;
+    if (this.get('hasMultipleBars')) {
+      return _.map(this.get('groupedData'), function(values, barName) {
+        posTop = 0;
+        negBottom = 0;
         stackedValues = _.map(values, function(d) {
+          var yMin, yMax;
+          if (d.value < 0) {
+            yMax = negBottom;
+            negBottom += d.value;
+            yMin = negBottom;
+          } else {
+            yMin = posTop;
+            posTop += d.value;
+            yMax = posTop;
+          }
+
           return {
-            y0: y0,
-            y1: y0 += Math.max(d.value, 0),
+            yMin: yMin,
+            yMax: yMax,
             value: d.value,
-            group: d.group,
-            label: d.label,
+            barLabel: d.barLabel,
+            sliceLabel: d.sliceLabel,
             color: d.color
           };
         });
 
         return {
-          group: groupName,
+          barLabel: barName,
           values: values,
           stackedValues: stackedValues,
-          totalValue: y0
+          max: posTop,
+          min: negBottom
         };
       });
 
     } else {
-      debugger;
-      if (Ember.isEmpty(this.get('data'))) {
+      if (Ember.isEmpty(this.get('filteredData'))) {
         return Ember.A();
       }
-      // If we do not have grouped data and are drawing stacked bars, keep the
-      // data in one group so it gets stacked
-      y0 = 0;
-      stackedValues = _.map(this.get('data'), function(d) {
+
+      posTop = 0;
+      negBottom = 0;
+      stackedValues = _.map(this.get('filteredData'), function(d) {
+        var yMin, yMax;
+        if (d.value < 0) {
+          yMax = negBottom;
+          negBottom += d.value;
+          yMin = negBottom;
+        } else {
+          yMin = posTop;
+          posTop += d.value;
+          yMax = posTop;
+        }
+
         return {
-          y0: y0,
-          y1: y0 += Math.max(d.value, 0)
+          yMin: yMin,
+          yMax: yMax,
+          value: d.value,
+          barLabel: d.barLabel,
+          sliceLabel: d.sliceLabel,
+          color: d.color
         };
       });
 
       return Ember.A([{
-        group: this.get('data.firstObject.group'),
-        values: this.get('data'),
+        barLabel: this.get('data.firstObject.barLabel'),
+        values: this.get('filteredData'),
         stackedValues: stackedValues,
-        totalValue: y0
+        max: posTop,
+        min: negBottom
       }]);
-
     }
-  // TODO(tony): Need to have stacked bars as a dependency here and the
-  // calculation be outside of this
   }),
 
   // ----------------------------------------------------------------------------
   // Layout
   // ----------------------------------------------------------------------------
 
-  labelHeightOffset: Ember.computed('_shouldRotateLabels', 'maxLabelHeight', 'labelHeight',
-    'labelPadding', function() {
+  labelHeightOffset: Ember.computed('_shouldRotateLabels', 'maxLabelHeight',
```

Comment:
Since you modify this, it would be a great if you can add some documentation for it.